qiskit_alice_bob_provider/local/translation_plugin.py

```diff
@@ -165,3 +180,117 @@ def pass_manager(
                 custom_pm.append(p)
 
         return custom_pm
+
+
+@lru_cache(maxsize=1)
+def _memoized_basic_approximations(
+    basis_gates: FrozenSet[str], depth: int
+) -> List[GateSequence]:
+    """Generating approximations for Solovay-Kitaev is costy: this function
+    caches those approximations"""
+    return generate_basic_approximations(basis_gates=basis_gates, depth=depth)
+
+
+class LocalLogicalCatPlugin(PassManagerStagePlugin):
+    """This plugin configures the Solavay-Kitaev synthesis for the special
+    case of logical qubits made out of physical cat qubits.
+
+    Here's what it does:
+    * Compute the basis gates to be used for unitary synthesis. This is
+      actually different from the basis gates supported by the backend: not
+      all backend basis gates can be used as basis gates for the SK synthesis.
+      Unfortunately, Qiskit does not allow setting a different basis gate
+      set for te synthesis step, so we hack our way around it.
+    * Compute approximations for the SK synthesis using only the 1-qubit gates
+      of the SK basis gate set
+    * Mix the SK synthesis with the transpilation passes from
+      LocalStatePreparationPlugin
+
+    This plugin wouldn't exist if Qiskit's transpile exposed more options to
+    configure the synthesis method (and didn't override the available options
+    in strange ways)."""
+
+    def pass_manager(
+        self,
+        pass_manager_config: PassManagerConfig,
+        optimization_level=None,
+    ) -> PassManager:
+        # Compute the discrete basis gates for the Solovay-Kitaev synthesis
+        proc = LogicalCatProcessor()
+        discrete_basis_gates: Set[str] = set()
+        discrete_1q_basis_gates: Set[str] = set()
+        for instr in proc.all_instructions():
+            qiskit_instr = processor_to_qiskit_instruction(instr)
+            if (
+                len(qiskit_instr.params) != 0
+                or 'measure' in qiskit_instr.name
+                or qiskit_instr.name == 'delay'
```

Comment:
What would you think of defining an array above with the qiskit instructions to avoid as basis gates and use it here? We may win on some understandablity and make it easier to introduce new gates.

Reply:
Good idea, I implemented this. Thanks!